### ISSUE TITLE
Issue with "If-Modified-Since: Thu, 01 Jan 1970 00:00:00 GMT" and no/blank "If-None-Match"

### ISSUE THREAD

**Issue by mdurn** [slact/nchan#350]
Hi Leo,

I was updating nchan and noticed a change in behavior when subscribing to an empty channel (there have been no publications yet), when using a starting "If-Modified-Since" of "Thu, 01 Jan 1970 00:00:00 GMT" and no "If-None-Match" (or it's blank). Easy to work around (and I'll end up changing our client to not send a blank "If-None-Match" by default), but just wanted to bring it up in case you think it's worth fixing. Guessing it has to do with that datetime evaluating to 0.

**Problem:**
Subscribing to an empty channel with "If-Modified-Since" of "Thu, 01 Jan 1970 00:00:00 GMT" and no/blank "If-None-Match" results in no response when a message is published to the channel (the sub  request continues to stay open).

**Config details:**
- nginx version: openresty/1.9.7.4
- nginx config snippet:
```
location ~ /pub/(\w+)$ {
  access_log off;
  allow 127.0.0.1;
  allow 10.1.128.0/21;
  deny  all;
  nchan_publisher;
  nchan_channel_group chatroom;
  nchan_channel_id $1;
  nchan_message_timeout  24h;
  nchan_message_buffer_length 10;
}

location ~ /sub/(\w+)$ {
  access_log off;
  nchan_subscriber;
  nchan_longpoll_multipart_response raw;
  nchan_channel_group chatroom;
  nchan_channel_id $1;
  nchan_subscriber_first_message oldest;
}
```

**Previous behavior (tested at master commit b86ffc7):**
1) restart nginx
2) Start curl request:
```
curl -i --header "If-Modified-Since: Thu, 01 Jan 1970 00:00:00 GMT" --header "If-None-Match: " "https://my.pubsub.domain/sub/15"
```
3) Publish to the channel
4) curl responds with the published message

**Current behavior (tested in 1.1.3 and 1.1.4):**
1) restart nginx
2) Start curl request:
```
curl -i --header "If-Modified-Since: Thu, 01 Jan 1970 00:00:00 GMT" --header "If-None-Match: " "https://my.pubsub.domain/sub/15"
```
3) Publish to the channel
4) curl has no response
5) manually kill curl
6) do the same curl request again
7) curl responds this time with the published message

Thanks for all the awesome updates/fixes in the last few months!

**Comment by slact** [comment]
Well, sending 1970-1-1 as the first msg time isn't in spec, but you're certainly not the only one to do it. Incidentally, I don't actually want to allow message times <=0 because they have special meaning within Nchan. So thanks to you pointing this out, I'm now filtering out 1970-1-1 and below and treating them as requests with no message id. That's a backwards-compatibility and (future) security win.

Fixed in 54b01eb .

**Comment by mdurn** [comment]
Figured it wasn't the best use case, but thanks for addressing it! :)
